insonmnia/worker/server.go

```diff
@@ -371,18 +372,38 @@ func (m *Worker) cancelDealTasks(deal *pb.Deal) error {
 }
 
 type runningContainerInfo struct {
-	Description Description   `json:"description,omitempty"`
-	Cinfo       ContainerInfo `json:"cinfo,omitempty"`
-	Spec        pb.TaskSpec   `json:"spec,omitempty"`
+	Description descriptionMarshallable   `json:"description,omitempty"`
+	Cinfo       containerInfoMarshallable `json:"cinfo,omitempty"`
+	Spec        pb.TaskSpec               `json:"spec,omitempty"`
+}
+
+type descriptionMarshallable struct {
+	Description
+	Reference reference.Field `json:"Reference"`
```

Comment:
maybe just store reference.Field in description for simplicity?

Reply:
That would be handy if it helped getting rid of this whole new ugly ```descriptionMarshallable``` type, but it looks that the ```Networks``` field remains another obstacle in the way here, so in either way the type stays until the ```Networks\Networks()``` disambiguation is resolved.